test/db2.connection.test.js

```diff
@@ -50,3 +50,47 @@ function generateDSN(config) {
     ';CurrentSchema=' + config.schema;
   return dsn;
 }
+
+describe('lazyConnect', function() {
+  it('should skip connect phase (lazyConnect = true)', function(done) {
+    var dsConfig = {
+      host: 'invalid-hostname',
+      port: 80,
+      database: 'invalid-database',
+      username: 'invalid-username',
+      password: 'invalid-password',
+      lazyConnect: true,
+    };
+    var ds = getDS(dsConfig);
+
+    var errTimeout = setTimeout(function() {
+      done();
+    }, 2000);
+    ds.on('error', function(err) {
+      clearTimeout(errTimeout);
```

Comment:
@jannyHou clearTimeout calls errTimeout which calls done() then at line 71 you call done(err) again so it ends up being called twice which results in the failure.  I'm a little confused as to why you need the errTimeout function so maybe you can explain that a bit.